### PR TITLE
Spring order

### PR DESCRIPTION
**Pull request by keda-vlad**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java:36]:
```suggestion
    @Transactional
    @Override
```
What if your method fails in the middle of execution? Do not forget about transactions

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    public OrderDto placeAnOrder(Long userId, PlaceAnOrderDto placeAnOrderDto) {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java:37]: Is it possible to reduce complexity of this method? maybe extract some logical parts into private methods

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    @Override
+    public List<OrderDto> getAllOrders(Long id, Pageable pageable) {
+        return orderRepository.findAllByUserId(id, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long id, ChangeStatusDto changeStatusDto) {
+        Order order = orderRepository.findById(id).orElseThrow(() ->
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java:57]: What if user will specify id of an order which do not belong to him? You should validate not only access to the particular endpoint, but also access of a user to the particular resource (make sure that in other methods and services you don't forget about it)

File: src/main/resources/db/changelog/changes/12-create-order-table.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        - createTable:
+            tableName: order
+            columns:
+              - column:
+                  name: id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+              - column:
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/12-create-order-table.yaml]: add FK

File: src/main/resources/db/changelog/changes/13-create-order-item-table.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+              - column:
+                  name: order_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+              - column:
+                  name: book_id
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/13-create-order-item-table.yaml]: add FK

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                new EntityNotFoundException("Can't find order by id = " + id));
+        order.setStatus(Order.Status.valueOf(changeStatusDto.status()));
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    private Order buildOrder(
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: JFUI: Word build is often associated with pattern builder, buildOrder for me means that you are building order, not creating it. In this particular case it does not matter, but it is still confusing

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Transactional
+    @Override
+    public OrderDto placeAnOrder(Long userId, PlaceAnOrderDto placeAnOrderDto) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java:37]: Think about what else methods should be transactional

File: src/main/resources/db/changelog/changes/15-add-foreign-keys.yaml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            baseColumnNames: book_id
+            constraintName: fk_order_items_book_id
+            referencedTableName: books
+            referencedColumnNames: id
+        - addForeignKeyConstraint:
+            baseTableName: order_items
+            baseColumnNames: order_id
+            constraintName: fk_order_items_order_id
+            referencedTableName: orders
+            referencedColumnNames: id
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/15-add-foreign-keys.yaml]: Add empty line

File: src/main/resources/db/changelog/changes/13-create-order-item-table.yaml (hunk 5 above)
Comment (andrii-hoienko) [reply src/main/resources/db/changelog/changes/13-create-order-item-table.yaml]: not resolved

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ, I left minor comments

File: src/main/resources/db/changelog/changes/13-create-order-item-table.yaml (hunk 5 above)
Comment (keda-vlad) [reply src/main/resources/db/changelog/changes/13-create-order-item-table.yaml]: Added in 15-add-foreign-keys.yaml

**Review by vovasalyha — APPROVED** [review]
approving, but resolve merge conflicts

**Pull request merged**